### ISSUE TITLE
Less compilation extend 'mixin' has no matches

### ISSUE THREAD

**Issue by patruvlad** [magento/magento2#23619]
### Preconditions (*)

1. Magento version 2.3.1 - 2.3.2
2. Production mode

### Steps to reproduce (*)

1. Having nodejs v8 installed
2. Setting up vanilla Magento installation using composer
3. Run `npm install less@2.7.3`
4. Run `bin/magento setup:static-content:deploy -f -t Magento/luma en_US`
5. Run `node_modules/.bin/lessc var/view_preprocessed/pub/static/frontend/Magento/luma/en_US/css/styles-m.less > /dev/null`
6. Notice the output (what you see is the output to `stderr`, since we redirected `stdout` to `/dev/null`):
```
extend ' .abs-modal-overlay' has no matches
extend ' .abs-add-clearfix-desktop' has no matches
extend ' .abs-blocks-2columns' has no matches
extend ' .abs-forms-general-desktop' has no matches
extend ' .abs-action-reset' has no matches
extend ' .abs-no-display-desktop' has no matches
extend ' .clearer' has no matches
extend ' .abs-clearfix' has no matches
extend ' .abs-action-print' has no matches
```

### Expected result (*)

1. The compiled css should not have anything like the `extend 'mixin_name' has no matches`

### Actual result (*)

1. Errors during less compilation

**Comment by patruvlad** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @patruvlad. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @patruvlad, here is your [Magento instance](https://i-23619-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-23619-2-3-develop.instances.magento-community.engineering/admin](https://i-23619-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by engcom-Bravo** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @engcom-Bravo. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @engcom-Bravo, here is your [Magento instance](https://i-23619-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-23619-2-3-develop.instances.magento-community.engineering/admin](https://i-23619-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by engcom-Bravo** [comment]
Hi @patruvlad thank you for your report, I'm not able to reproduce issue by steps you described on clean 2.3-develop.

**Comment by patruvlad** [comment]
I modified the description with some Manual testing scenario , can this be reopened and rechecked?

**Comment by hostep** [comment]
Just to clarify, we've opened this issue, because it indicates that certain less code isn't following the less standard. It accidentally didn't output errors with the [less compiler](https://github.com/oyejorge/less.php/) Magento ships with but if we want to use an [alternative less compiler](https://github.com/less/less.js) which is more strict, it should still work.

Here are some findings:
- `.abs-action-reset` is being [defined](https://github.com/magento/magento2/blob/adb5dff9c2cf5b158df51f0efdf788047645b7c6/app/design/adminhtml/Magento/backend/web/css/source/_actions.less#L21-L23) in adminhtml theme, but is [used](https://github.com/magento/magento2/blob/adb5dff9c2cf5b158df51f0efdf788047645b7c6/app/design/frontend/Magento/luma/Magento_Checkout/web/css/source/module/checkout/fields/_file-uploader.less#L132) in a frontend theme, so that can never work. (problem was introduced recently in 2.3.2 by https://github.com/magento/magento2/commit/ffbf8e6da2f5#diff-3df2988da960a31971a90e02c78804b8)
- `.abs-action-print` is being [defined](https://github.com/magento/magento2/blob/adb5dff9c2cf5b158df51f0efdf788047645b7c6/app/design/frontend/Magento/luma/web/css/source/_extends.less#L1117-L1121) in a media query, but then [used](https://github.com/magento/magento2/blob/adb5dff9c2cf5b158df51f0efdf788047645b7c6/app/design/frontend/Magento/luma/Magento_Sales/web/css/source/_module.less#L10) in a different media query so that might cause this problem
- `..abs-forms-general-desktop`, same thing over here, is being [defined](https://github.com/magento/magento2/blob/adb5dff9c2cf5b158df51f0efdf788047645b7c6/app/design/frontend/Magento/luma/web/css/source/_extends.less#L829-L830) in a media query, but then [used](https://github.com/magento/magento2/blob/adb5dff9c2cf5b158df51f0efdf788047645b7c6/app/design/frontend/Magento/luma/Magento_CatalogSearch/web/css/source/_module.less#L20) in a different one, so the definition can't be found when not in the exact same media query I assume?
- ...

**Comment by engcom-Bravo** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @engcom-Bravo. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @engcom-Bravo, here is your [Magento instance](https://i-23619-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-23619-2-3-develop.instances.magento-community.engineering/admin](https://i-23619-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by engcom-Bravo** [comment]
@magento give me 2.2-develop instance

**Comment by magento-engcom-team** [comment]
Hi @engcom-Bravo. Thank you for your request. I'm working on Magento 2.2-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @engcom-Bravo, here is your [Magento instance](https://i-23619-2-2-develop.instances.magento-community.engineering).
Admin access: [https://i-23619-2-2-develop.instances.magento-community.engineering/admin](https://i-23619-2-2-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Bravo 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-18025`, `MC-18026` were created

**Issue Available**: @engcom-Bravo, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @patruvlad. Thank you for your report. 
The issue has been fixed in magento/magento2#24001 by @hostep in 2.3-develop branch
Related commit(s):
 - [cbcb8c6523cf4ac83e9a986cea305d74b8004a20](https://github.com/magento/magento2/commit/cbcb8c6523cf4ac83e9a986cea305d74b8004a20)
 - [ed987037d526fb9ea2a09d0345ffbd48afdee856](https://github.com/magento/magento2/commit/ed987037d526fb9ea2a09d0345ffbd48afdee856)
 - [beeb56c354467189f3ed9a4c6a5b4ea169b62846](https://github.com/magento/magento2/commit/beeb56c354467189f3ed9a4c6a5b4ea169b62846)
 - [d8c86b32907a753d58ac969c5315da3f6da64658](https://github.com/magento/magento2/commit/d8c86b32907a753d58ac969c5315da3f6da64658)
 - [937b2c048660d1f5c2e3ace2564edffb433f0fb4](https://github.com/magento/magento2/commit/937b2c048660d1f5c2e3ace2564edffb433f0fb4)

The fix will be available with the upcoming 2.3.4 release.

**Comment by magento-engcom-team** [comment]
Hi @patruvlad. Thank you for your report. 
The issue has been fixed in magento/magento2#24003 by @hostep in 2.4-develop branch
Related commit(s):
 - [820fd790d85c6ae00413e281689fd45613e2d63a](https://github.com/magento/magento2/commit/820fd790d85c6ae00413e281689fd45613e2d63a)
 - [164a0531b0efa3707f1192249e50469c493c96d3](https://github.com/magento/magento2/commit/164a0531b0efa3707f1192249e50469c493c96d3)
 - [8419d5ccf1c052cc92c5b1c8a61eedd96e2be5bb](https://github.com/magento/magento2/commit/8419d5ccf1c052cc92c5b1c8a61eedd96e2be5bb)
 - [4abaaa094f3c6f9696fc90a5e2471e873fff816d](https://github.com/magento/magento2/commit/4abaaa094f3c6f9696fc90a5e2471e873fff816d)
 - [5657076c219ea5024fdf90d40e8ad573ad24a7c6](https://github.com/magento/magento2/commit/5657076c219ea5024fdf90d40e8ad573ad24a7c6)

The fix will be available with the upcoming 2.4.0 release.
